File: src/main/java/ru/protei/oxmldoc/style/NumberFormat.java
package ru.protei.oxmldoc.style;

public class NumberFormat {
    private final int id;
    private String format;

    public NumberFormat(int id) {
        this.id = id;
    }

    private static StringBuilder mappend(StringBuilder sb, char appendedChar, int times) {
        while (times > 0) {
            sb.append(appendedChar);
            times--;
        }
        return sb;
    }

    public int getId() {
        return id;
    }

    public String getFormat() {
        return format;
    }

    public void setFormat(String format) {
        this.format = format;
    }

    public NumberFormat asNumberFmt(int fractionDigits) {
        return asNumberFmt(1, fractionDigits);
    }

    private NumberFormat asNumberFmt(int integerDigits, int fractionDigits) {
        StringBuilder sb = mappend(new StringBuilder(), '0', integerDigits).append('.');
        this.format = mappend(sb, '0', fractionDigits).toString();
        return this;
    }
}

File: src/main/kotlin/ru/protei/oxmldoc/style/NumberFormat.kt
package ru.protei.oxmldoc.style

import java.util.stream.IntStream

class NumberFormat(val id: Int, var format: String){

    fun asNumberFormat(fractionDigits: Int): NumberFormat {
        return asNumberFormat(1, fractionDigits)
    }

    fun asNumberFormat(integerDigits: Int, fractionDigits: Int): NumberFormat {
        val sb = append(StringBuilder(), '0', integerDigits).append('.')
        format = append(sb, '0', fractionDigits).toString()
        return this
    }

    private fun append(sb: StringBuilder, char: Char, times: Int): StringBuilder {
        (0 .. times).forEach { sb.append(char) }
        return sb
    }
}